### PR TITLE
Add debug-stackoverflow.md

### PR DESCRIPTION
**Pull request by tommcdon**
Add tutorial on debugging a stack overflow 

Contributes to https://github.com/dotnet/diagnostics/issues/515

### CODE REVIEW

File: docs/core/diagnostics/debug-stackoverflow.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    ````
+    > export COMPlus_DbgEnableMiniDump=1
+    > dotnet run
+    Stack overflow.
+    Writing minidump with heap to file /tmp/coredump.6412
+    Written 58191872 bytes (14207 pages) to core file
+    ````
+
+2. Debug the dump in lldb to see the failing stack
```
Comment (sywhang) [inline docs/core/diagnostics/debug-stackoverflow.md:54]: Before we open it up with lldb, should we add a step here that tells them to install SOS with a link to dotnet-sos?

File: docs/core/diagnostics/debug-stackoverflow.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    Version History:
+      ILCodeVersion:      0000000000000000
+      ReJIT ID:           0
+      IL Addr:            0000000000000000
+         CodeAddr:           00007f59b40900a0  (MinOptJitted)
+         NativeCodeVersion:  0000000000000000
+    Source file:  /temp/Program.cs @ 9
+    ````
+
+4. Go look at the indicated method temp.Program.Main(System.String[]) and  source "/temp/Program.cs @ 9" to see if you can figure out what you did wrong. If it still wasn't clear you could add logging in that area of the code.
```
Comment (sywhang) [inline docs/core/diagnostics/debug-stackoverflow.md:93]: nit: extra space
```suggestion
4. Go look at the indicated method temp.Program.Main(System.String[]) and source "/temp/Program.cs @ 9" to see if you can figure out what you did wrong. If it still wasn't clear you could add logging in that area of the code.
```

File: docs/core/diagnostics/debug-stackoverflow.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        frame #261933: 0x00007f59b40900cc
+        frame #261934: 0x00007f59b40900cc
+        frame #261935: 0x00007f5a2d4a080f libcoreclr.so`CallDescrWorkerInternal at unixasmmacrosamd64.inc:867
+        frame #261936: 0x00007f5a2d3cc4c3 libcoreclr.so`MethodDescCallSite::CallTargetWorker(unsigned long const*, unsigned long*, int) at callhelpers.cpp:70
+        frame #261937: 0x00007f5a2d3cc468 libcoreclr.so`MethodDescCallSite::CallTargetWorker(this=<unavailable>, pArguments=0x00007ffe8222e7b0, pReturnValue=0x0000000000000000, cbReturnValue=0) at callhelpers.cpp:604
+        frame #261938: 0x00007f5a2d4b6182 libcoreclr.so`RunMain(MethodDesc*, short, int*, PtrArray**) [inlined] MethodDescCallSite::Call(this=<unavailable>, pArguments=<unavailable>) at callhelpers.h:468
+    ...
+    ````
+
+3. Use the SOS ip2md command to figure out what method is repeating in all those mystery frames
```
Comment (sywhang) [inline docs/core/diagnostics/debug-stackoverflow.md:72]: Can we add somewhere here 0x00007f59b40900cc is being repeated many times in the stack above? It's obvious but just to highlight it :)

File: docs/core/diagnostics/index.md [hunk 4]
```diff
@@ -79,6 +79,10 @@ The [dotnet-gcdump](dotnet-gcdump.md) tool is a way to collect GC (Garbage Colle
 
 [Tutorial: Debug deadlock](debug-deadlock.md) shows you how to use the [dotnet-dump](dotnet-dump.md) tool to investigate threads and locks.
 
+### Debug a StackOverflow
+
+[Tutorial: Debug a StackOverflow](debug-stackoverflow.md) demonstrates how to debug a <xref:System.StackOverflowException> on Linux.
+
 ### Debug Linux dumps
```
Comment (sywhang) [inline docs/core/diagnostics/index.md:87]: Can we also add this in toc.yml? Should be under docs/fundamentals/toc.yml

File: docs/core/diagnostics/debug-stackoverflow.md (hunk 1 above)
Comment (tommcdon) [reply docs/core/diagnostics/debug-stackoverflow.md]: Great idea!  I've pushed an update for this.

File: docs/core/diagnostics/debug-stackoverflow.md (hunk 3 above)
Comment (tommcdon) [reply docs/core/diagnostics/debug-stackoverflow.md]: good suggestion, I've pushed a change

File: docs/core/diagnostics/index.md (hunk 4 above)
Comment (tommcdon) [reply docs/core/diagnostics/index.md:87]: Done

**Review by sywhang — APPROVED** [review]

**Pull request merged**